src/Microsoft.Health.Api/Features/Audit/AuditEventTypeMapping.cs

```diff
@@ -74,6 +74,7 @@ public Task StartAsync(CancellationToken cancellationToken)
                     return (ad.ControllerName, ad.ActionName, Attribute: attribute);
                 })
                 .Where(item => item.Attribute != null)
+                .Distinct()
```

Comment:
I think there may be some strange cases where this doesn't work as expected. In particular, I am unsure what the value for `ActionName` will be if a controller uses multiple overloads of the same method name for its actions:

eg.
```csharp
public class ExampleController : Controller
{
    [Route("/Examples/{number}")]
    [AuditEventTypeAttribute("Get-By-Number")]
    public async Task GetAsync(int number) { ... }

    [Route("/Examples/{dt}")]
    [AuditEventTypeAttribute("Get-By-DateTime")]
    public async Task GetAsync(DateTime dt) { ... }
}
```

If they're the same, then I could image their respective tuples will not be filtered by `Distinct()` and could cause the same exception (presumably `ArgumentException`?). So, I think we need to push the de-duplication up a bit and perhaps throw an error when we get different attributes for the same `(ControllerName, ActionName)` tuple. That way we prevent unexpected behavior.

Of course, if this scenario is not possible then ignore me lol.

Reply:
Hm, good point! I just double-checked and you are right, Distinct() will not remove duplicate keys with different values. 

The existing code will throw an ArgumentException in this case. I guess we would want a more descriptive error message instead, so people know to pick a unique method name when they run into this error?

Reply:
Updated to throw a specific error for this case

Reply:
I began writing comments to explain that we're not handling the case where a developer annotates different actions with the same `AuditEventType`... but that scenario sounds out-of-scope at the moment. Perhaps just a `TODO` will suffice lol

Reply:
Added a TODO